codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/SymbolVisitor.java

```diff
@@ -0,0 +1,518 @@
+                .build();
+    }
+
+    @Override
+    public Symbol floatShape(FloatShape shape) {
+        return symbolBuilderFor(shape, "float32")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .build();
+    }
+
+    @Override
+    public Symbol documentShape(DocumentShape shape) {
+        return null;
+    }
+
+    @Override
+    public Symbol doubleShape(DoubleShape shape) {
+        return symbolBuilderFor(shape, "float64")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .build();
+    }
+
+    @Override
+    public Symbol bigIntegerShape(BigIntegerShape shape) {
+        return createBigSymbol(shape, "Int");
+    }
+
+    @Override
+    public Symbol bigDecimalShape(BigDecimalShape shape) {
+
+        return createBigSymbol(shape, "Float");
+    }
+
+    private Symbol createBigSymbol(Shape shape, String symbolName) {
+        return symbolBuilderFor(shape, symbolName, SmithyGoDependency.BIG)
+                .build();
+    }
+
+    @Override
+    public Symbol operationShape(OperationShape shape) {
+        String name = getDefaultShapeName(shape);
+        return SymbolUtils.createPointableSymbolBuilder(shape, name, rootModuleName)
+                .definitionFile(String.format("./api_op_%s.go", name))
+                       .addDependency("./types", "./types")
+                .build();
+    }
+
+    @Override
+    public Symbol resourceShape(ResourceShape shape) {
+        // TODO: implement resources
+        return SymbolUtils.createPointableSymbolBuilder(shape, "nil").build();
+    }
+
+    @Override
+    public Symbol serviceShape(ServiceShape shape) {
+        return symbolBuilderFor(shape, "Client", rootModuleName)
+                .definitionFile("./api_client.go")
+                .build();
+    }
+
+    @Override
+    public Symbol stringShape(StringShape shape) {
+        if (shape.hasTrait(EnumTrait.class)) {
+            String name = getDefaultShapeName(shape);
+            return symbolBuilderFor(shape, name, typesPackageName)
+                    .definitionFile("./types/enums.go")
+                    .build();
+        }
+
+        return symbolBuilderFor(shape, "string")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .build();
+    }
+
+    @Override
+    public Symbol structureShape(StructureShape shape) {
+        String name = getDefaultShapeName(shape);
+        String typeName = OperationIndex.of(model).isOutputStructure(shape) ? "Result" : name;
+//        if (shape.getId().getNamespace().equals(CodegenUtils.getSyntheticTypeNamespace())) {
+//            Optional<String> boundOperationName = getNameOfBoundOperation(shape);
+//            if (boundOperationName.isPresent()) {
+//                return symbolBuilderFor(shape, name, rootModuleName)
+//                        .definitionFile("./api_op_" + boundOperationName.get() + ".go")
+//                        .build();
+//            }
+//        }
```

Comment:
Did we comment this out or did Smithy-Go's authors?
If it was us, why?

Also, this looks like it may be here to support Void operations...
though I am not sure...

Reply:
This was for UTs, uncommented this.